### PR TITLE
refactor: refactor wasm storage module collections

### PR DESCRIPTION
**Pull request by hacheigriega**
Fixing some issues introduced during the migration to collections. 

- Fix the issue where we add wasm type prefix unnecessarily.
- Update field names so they are clearer.
- Remove unnecessary functions.

### CODE REVIEW

**Comment by Thomasvdam** [comment]
This also includes a bunch of the tally VM stuff, is that intended?

**Comment by hacheigriega** [comment]
> This also includes a bunch of the tally VM stuff, is that intended?

removed the tally vm import, which is not relevant here!

**Review by DeshErBojhaa — APPROVED** [review]
🎉 Great work!!

File: x/wasm-storage/keeper/test_utils/debug.wasm
Comment (Thomasvdam) [inline x/wasm-storage/keeper/test_utils/debug.wasm:1]: Is this not also part of the tally?

**Pull request merged**